File: 1895-minimum-number-of-operations-to-move-all-balls-to-each-box/1895-minimum-number-of-operations-to-move-all-balls-to-each-box.py
class Solution(object):
    def minOperations(self, boxes):
        """
        :type boxes: str
        :rtype: List[int]
        """
        n = len(boxes)
        lis = [0] * len(boxes)
        pf = [0] * (n + 1)
        pfe = [0] * (n + 1)
        suf = [0] * (n + 1)
        sufe = [0] * (n + 1)

        for i in range(0, n):
            if boxes[i] == '1':
                pf[i + 1] = pf[i] + pfe[i] + 1
                pfe[i + 1] = pfe[i] + 1
            else:
                pf[i + 1] = pf[i] + pfe[i]
                pfe[i + 1] = pfe[i]

        for i in range(n - 1, -1, -1):
            if boxes[i] == '1':
                suf[i] = suf[i + 1] + sufe[i + 1] + 1
                sufe[i] = sufe[i + 1] + 1
            else:
                suf[i] = suf[i + 1] + sufe[i + 1]
                sufe[i] = sufe[i + 1]

        for i in range(0, n):
            lis[i] = pf[i] + suf[i + 1]

        return lis

File: 1895-minimum-number-of-operations-to-move-all-balls-to-each-box/1895-minimum-number-of-operations-to-move-all-balls-to-each-box.cpp
class Solution {
public:
    vector<int> minOperations(string boxes) {
        vector<int> ans(boxes.size(), 0);

        for(int i = 0; i < boxes.size(); i++) {
            int cnt = 0;
            for(int j = 0; j < boxes.size(); j++) {
                if(i != j && boxes[j] == '1') {
                    cnt += abs(j - i);
                }
            }
            ans[i] = cnt;
        }

        return ans;
    }
};